packages/cli/src/services/abstract-check-runner.ts

```diff
@@ -203,9 +201,19 @@ export default abstract class AbstractCheckRunner extends EventEmitter {
   }
 
   private startSchedulingDelayTimeout () {
-    this.timeouts.set(SCHEDULING_DELAY_EXCEEDED_TIMEOUT_KEY, setTimeout(() =>
-      this.emit(Events.MAX_SCHEDULING_DELAY_EXCEEDED), DEFAULT_SCHEDULING_DELAY_EXCEEDED_MS,
-    ))
+    let scheduledCheckCount = 0
+    const numChecks = this.checks.size
+    if (numChecks === 0) {
+      return
+    }
+    const schedulingDelayExceededTimeout = setTimeout(
+      () => this.emit(Events.MAX_SCHEDULING_DELAY_EXCEEDED),
+      DEFAULT_SCHEDULING_DELAY_EXCEEDED_MS,
+    )
```

Comment:
We should also clear this in the exception try catch

Reply:
Good catch, I'll update it accordingly